ArduPlane/commands_logic.cpp

```diff
@@ -726,8 +726,10 @@ bool Plane::verify_loiter_turns(const AP_Mission::Mission_Command &cmd)
     // LOITER_TURNS makes no sense as VTOL
     auto_state.vtol_loiter = false;
 
-    if (condition_value != 0) {
-        // primary goal, loiter time
+    if (!reached_loiter_target()) {
```

Comment:
This can be a little tricky. If you set a new loiter target and your already loitering `reached_loiter_target` doesn't go false until the nav controller has run on the new target. It would be worth checking what happens with two consecutive loiter turns way points.

Reply:
This new autotest already contains two consecutive zero turn loiters. 

AP had a bug that was causing issues with the RTL AUTOLAND. When it was configured to fly home first. If you were already on a loiter, it would skip the fly home loiter because it had already deemed as reaching the loiter target. 

See: https://github.com/ArduPilot/ardupilot/pull/26413

I suspect this is the behaviour you're eluding to?

Reply:
I would need to check later today, but Im leaning towards that fix actually being the reason this fix works.

Reply:
Now I revisit that fix, it is explicitly for mode RTL.
I've modified the autotest to add in a guided orbit, and this fix works as expected too.

![image](https://github.com/user-attachments/assets/38a016de-28dd-4728-a794-45e65ee80b9c)

Reply:
The same function is called in `verify_loiter_time()` in much the same way.

I'll bring this to the dev call for broader discussion.

Reply:
I'm convinced. 

BTW, you could early return false. 

```suggestion
    if (!reached_loiter_target()) {
        return false;
    }
```